### ISSUE TITLE
softmax_cross_entropy_with_logits_v2 doesn't support hypergradients of labels

### ISSUE THREAD

**Issue by fps7806** [tensorflow/tensorflow#26495]
Current implementation of tf.nn.softmax_cross_entropy_with_logits_v2 doesn't correctly implement the  hypergradients w.r.t labels. I implemented a hypergradient demo where I used softmax_cross_entropy_with_logits_v2 to compute meta_loss, and a custom function to compute meta_loss2. We should either expect that:
```
tf.gradients(meta_loss, [a]) == tf.gradients(meta_loss2, [a])
```

Or that 
```
tf.gradients(meta_loss, [a]) == None
```

Instead, TensorFlow incorrectly returns 0s for hypergradients of softmax_cross_entropy_with_logits_v2 w.r.t labels.

Code to reproduce:
```
import tensorflow as tf


a = tf.get_variable("a", (2, 2)) * 10
b = tf.get_variable("b", (2, 2)) * 10
a2 = tf.get_variable("a2", (2, 2)) * 10

labels = tf.nn.softmax(a, -1)
labels2 = tf.nn.softmax(a2, -1)
loss = tf.reduce_sum(tf.nn.softmax_cross_entropy_with_logits_v2(labels=labels, logits=b))
grads = tf.gradients(loss, [b])

b_tp1 = b + grads[0]
meta_loss = tf.nn.softmax_cross_entropy_with_logits_v2(labels=labels2, logits=b_tp1)

logits = tf.nn.log_softmax(b, -1)
loss2 = tf.reduce_sum(-tf.reduce_sum(labels * logits))
grads2 = tf.gradients(loss2, [b])
b2_tp1 = b + grads2[0]

meta_loss2 = tf.nn.softmax_cross_entropy_with_logits_v2(labels=labels2, logits=b2_tp1)

sess = tf.Session()
sess.run(tf.global_variables_initializer())
print(sess.run([tf.gradients(meta_loss, [a]), tf.gradients(meta_loss2, [a])]))
```

**System information**
- TensorFlow-gpu 1.10.1

**Comment by jvishnuvardhan** [comment]
@fps7806 Could you check whether the bug persists with latest TF? Thanks!

**Comment by fps7806** [comment]
> @fps7806 Could you check whether the bug persists with latest TF? Thanks!

Still present on the latest release (1.13.1)

**Comment by jvishnuvardhan** [comment]
@fps7806 Is this still an issue for you. Currently `TF1.x` is not supported. Can you please check with `TF2.x` and let us know whether the issue persists with `TF2.x`. Thanks!
